crates/astria-conductor/src/executor.rs

```diff
@@ -521,32 +570,27 @@ mod test {
             .await
             .unwrap();
 
-        // should not have executed or finalized the block
-        assert!(finalized_blocks.lock().await.is_empty());
-        assert!(
-            finalized_blocks
-                .lock()
-                .await
-                .get(&executor.execution_state)
-                .is_none()
-        );
         assert_eq!(previous_execution_state, executor.execution_state,);
         // should be empty because nothing was executed
         assert!(executor.sequencer_hash_to_execution_hash.is_empty());
     }
 
     #[tokio::test]
     async fn execute_unexecuted_da_block_with_no_transactions() {
-        let conf = get_test_config();
-        let chain_id = ChainId::new(conf.chain_id.as_bytes().to_vec()).unwrap();
-        let finalized_blocks = Arc::new(Mutex::new(HashSet::new()));
-        let execution_client = MockExecutionClient {
-            finalized_blocks: finalized_blocks.clone(),
-        };
-        let (mut executor, _) = Executor::new(
-            execution_client,
+        let mut cfg = get_test_config();
+        let chain_id = ChainId::new(cfg.chain_id.as_bytes().to_vec()).unwrap();
+
+        let execution_server = MockExecution::spawn().await;
+        cfg.execution_rpc_url = execution_server.local_addr();
+
+        let (_block_tx, block_rx) = mpsc::unbounded_channel();
+        let (_shutdown_tx, shutdown_rx) = oneshot::channel();
+        let mut executor = Executor::new(
+            &cfg.execution_rpc_url,
             chain_id,
-            conf.disable_empty_block_execution,
+            cfg.disable_empty_block_execution,
+            block_rx,
+            shutdown_rx,
```

Comment:
the setup for these tests appears to be basically the same in many of them, maybe create a helper func for this?

Reply:
Moved all of that into a single setup fn.